From 8bff3e06c4bea899bf2e54fa77184e4bb70b76e2 Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@tq-group.com>
Date: Wed, 17 Oct 2018 18:22:00 +0200
Subject: [PATCH 1/2] aarch64: dt: rewrite trees for TQMa8QX

- split module and baseboard
- cleanup and remove not implemented interfaces

Signed-off-by: Markus Niebel <Markus.Niebel@tq-group.com>
---
 arch/arm64/boot/dts/freescale/Makefile        |   4 +-
 .../fsl-imx8qxp-tqma8qx-a0-mba8qx.dts         |  20 ++
 ...8qx-a0.dts => fsl-imx8qxp-tqma8qx-a0.dtsi} |   2 -
 .../freescale/fsl-imx8qxp-tqma8qx-mba8qx.dts  |  19 ++
 .../freescale/fsl-imx8qxp-tqma8qx-mba8qx.dtsi | 125 +++++++++++++
 ...p-tqma8qx.dts => fsl-imx8qxp-tqma8qx.dtsi} | 173 +-----------------
 6 files changed, 167 insertions(+), 176 deletions(-)
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8qxp-tqma8qx-a0-mba8qx.dts
 rename arch/arm64/boot/dts/freescale/{fsl-imx8qxp-tqma8qx-a0.dts => fsl-imx8qxp-tqma8qx-a0.dtsi} (94%)
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8qxp-tqma8qx-mba8qx.dts
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8qxp-tqma8qx-mba8qx.dtsi
 rename arch/arm64/boot/dts/freescale/{fsl-imx8qxp-tqma8qx.dts => fsl-imx8qxp-tqma8qx.dtsi} (50%)

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index b0bcb92276db..e7e677ab8f53 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -54,8 +54,8 @@ dtb-$(CONFIG_ARCH_FSL_IMX8QXP) += fsl-imx8qxp-lpddr4-arm2.dtb \
 				  fsl-imx8qxp-lpddr4-arm2-a0.dtb \
 				  fsl-imx8dx-lpddr4-arm2.dtb \
 				  fsl-imx8dxp-lpddr4-arm2.dtb \
-				  fsl-imx8qxp-tqma8qx.dtb \
-				  fsl-imx8qxp-tqma8qx-a0.dtb
+				  fsl-imx8qxp-tqma8qx-mba8qx.dtb \
+				  fsl-imx8qxp-tqma8qx-a0-mba8qx.dtb
 dtb-$(CONFIG_ARCH_FSL_IMX8MQ) += fsl-imx8mq-ddr3l-arm2.dtb \
 				 fsl-imx8mq-ddr4-arm2.dtb \
 				 fsl-imx8mq-ddr4-arm2-gpmi-nand.dtb \
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-tqma8qx-a0-mba8qx.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-tqma8qx-a0-mba8qx.dts
new file mode 100644
index 000000000000..b778f127f53d
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-tqma8qx-a0-mba8qx.dts
@@ -0,0 +1,20 @@
+/*
+ * Copyright 2018 TQ Systems GmbH
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+/dts-v1/;
+
+#include "fsl-imx8qxp-tqma8qx.dtsi"
+#include "fsl-imx8qxp-tqma8qx-a0.dtsi"
+#include "fsl-imx8qxp-tqma8qx-mba8qx.dtsi"
+
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-tqma8qx-a0.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-tqma8qx-a0.dtsi
similarity index 94%
rename from arch/arm64/boot/dts/freescale/fsl-imx8qxp-tqma8qx-a0.dts
rename to arch/arm64/boot/dts/freescale/fsl-imx8qxp-tqma8qx-a0.dtsi
index b9bdf0313ee1..e93d3cbec9be 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-tqma8qx-a0.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-tqma8qx-a0.dtsi
@@ -12,8 +12,6 @@
  * GNU General Public License for more details.
  */
 
-#include "fsl-imx8qxp-tqma8qx.dts"
-
 &vpu_encoder {
 	status = "disabled";
 };
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-tqma8qx-mba8qx.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-tqma8qx-mba8qx.dts
new file mode 100644
index 000000000000..7e991f4cc73b
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-tqma8qx-mba8qx.dts
@@ -0,0 +1,19 @@
+/*
+ * Copyright 2018 TQ Systems GmbH
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+/dts-v1/;
+
+#include "fsl-imx8qxp-tqma8qx.dtsi"
+#include "fsl-imx8qxp-tqma8qx-mba8qx.dtsi"
+
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-tqma8qx-mba8qx.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-tqma8qx-mba8qx.dtsi
new file mode 100644
index 000000000000..14bfd2ff4c36
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-tqma8qx-mba8qx.dtsi
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2018 TQ Systems GmbH
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include <dt-bindings/net/ti-dp83867.h>
+
+/ {
+	model = "TQ Systems i.MX8QXP TQMa8xX on MBa8xX";
+	compatible = "tqc,tqma8qx-mba8qx", "tqc,tqma8qx", "fsl,imx8qxp";
+
+	chosen {
+		bootargs = "console=ttyLP1,115200 earlycon=lpuart32,0x5a070000,115200";
+		stdout-path = &lpuart1;
+	};
+
+	regulators {
+		compatible = "simple-bus";
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		reg_usdhc2_vmmc: usdhc2_vmmc {
+			compatible = "regulator-fixed";
+			regulator-name = "SD1_SPWR";
+			regulator-min-microvolt = <3000000>;
+			regulator-max-microvolt = <3000000>;
+			gpio = <&gpio4 19 GPIO_ACTIVE_HIGH>;
+			enable-active-high;
+			startup-delay-us = <100>;
+			off-on-delay-us = <12000>;
+		};
+	};
+};
+
+&iomuxc {
+	pinctrl-names = "default";
+	tqma8qx-mba8qx {
+
+		pinctrl_lpuart0: lpuart0grp {
+			fsl,pins = <
+				SC_P_UART0_RX_ADMA_UART0_RX	0x06000020
+				SC_P_UART0_TX_ADMA_UART0_TX	0x06000020
+			>;
+		};
+
+		pinctrl_lpuart1: lpuart1grp {
+			fsl,pins = <
+				SC_P_UART1_RX_ADMA_UART1_RX	0x06000020
+				SC_P_UART1_TX_ADMA_UART1_TX	0x06000020
+			>;
+		};
+
+
+		pinctrl_usdhc2_gpio: usdhc2gpiogrp {
+			fsl,pins = <
+				SC_P_USDHC1_RESET_B_LSIO_GPIO4_IO19	0x06000021
+				SC_P_USDHC1_WP_LSIO_GPIO4_IO21		0x06000021
+				SC_P_USDHC1_CD_B_LSIO_GPIO4_IO22	0x06000021
+			>;
+		};
+
+		pinctrl_usdhc2: usdhc2grp {
+			fsl,pins = <
+				SC_P_USDHC1_CLK_CONN_USDHC1_CLK		0x06000041
+				SC_P_USDHC1_CMD_CONN_USDHC1_CMD		0x06000021
+				SC_P_USDHC1_DATA0_CONN_USDHC1_DATA0	0x06000021
+				SC_P_USDHC1_DATA1_CONN_USDHC1_DATA1	0x06000021
+				SC_P_USDHC1_DATA2_CONN_USDHC1_DATA2	0x06000021
+				SC_P_USDHC1_DATA3_CONN_USDHC1_DATA3	0x06000021
+				SC_P_USDHC1_VSELECT_CONN_USDHC1_VSELECT	0x06000021
+			>;
+		};
+
+		pinctrl_usdhc2_100mhz: usdhc2grp100mhz {
+			fsl,pins = <
+				SC_P_USDHC1_CLK_CONN_USDHC1_CLK		0x06000040
+				SC_P_USDHC1_CMD_CONN_USDHC1_CMD		0x06000020
+				SC_P_USDHC1_DATA0_CONN_USDHC1_DATA0	0x06000020
+				SC_P_USDHC1_DATA1_CONN_USDHC1_DATA1	0x06000020
+				SC_P_USDHC1_DATA2_CONN_USDHC1_DATA2	0x06000020
+				SC_P_USDHC1_DATA3_CONN_USDHC1_DATA3	0x06000020
+				SC_P_USDHC1_VSELECT_CONN_USDHC1_VSELECT	0x06000020
+			>;
+		};
+
+		pinctrl_usdhc2_200mhz: usdhc2grp200mhz {
+			fsl,pins = <
+				SC_P_USDHC1_CLK_CONN_USDHC1_CLK		0x06000040
+				SC_P_USDHC1_CMD_CONN_USDHC1_CMD		0x06000020
+				SC_P_USDHC1_DATA0_CONN_USDHC1_DATA0	0x06000020
+				SC_P_USDHC1_DATA1_CONN_USDHC1_DATA1	0x06000020
+				SC_P_USDHC1_DATA2_CONN_USDHC1_DATA2	0x06000020
+				SC_P_USDHC1_DATA3_CONN_USDHC1_DATA3	0x06000020
+				SC_P_USDHC1_VSELECT_CONN_USDHC1_VSELECT	0x06000020
+			>;
+		};
+	};
+};
+
+&lpuart1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_lpuart1>;
+	status = "okay";
+};
+
+&usdhc2 {
+	pinctrl-names = "default", "state_100mhz", "state_200mhz";
+	pinctrl-0 = <&pinctrl_usdhc2>, <&pinctrl_usdhc2_gpio>;
+	pinctrl-1 = <&pinctrl_usdhc2_100mhz>, <&pinctrl_usdhc2_gpio>;
+	pinctrl-2 = <&pinctrl_usdhc2_200mhz>, <&pinctrl_usdhc2_gpio>;
+	bus-width = <4>;
+	cd-gpios = <&gpio4 22 GPIO_ACTIVE_LOW>;
+	wp-gpios = <&gpio4 21 GPIO_ACTIVE_HIGH>;
+	vmmc-supply = <&reg_usdhc2_vmmc>;
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-tqma8qx.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-tqma8qx.dtsi
similarity index 50%
rename from arch/arm64/boot/dts/freescale/fsl-imx8qxp-tqma8qx.dts
rename to arch/arm64/boot/dts/freescale/fsl-imx8qxp-tqma8qx.dtsi
index 72066f70f6e2..d3b9397184ee 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-tqma8qx.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-tqma8qx.dtsi
@@ -12,8 +12,6 @@
  * GNU General Public License for more details.
  */
 
-/dts-v1/;
-
 /* First 128KB is for PSCI ATF. */
 /* Last 127M is for M4/RPMSG */
 /memreserve/ 0x80000000 0x08000000;
@@ -24,11 +22,6 @@
 	model = "TQ Systems i.MX8QXP TQMa8xX";
 	compatible = "fsl,tqma8qx", "fsl,imx8qxp";
 
-	chosen {
-		bootargs = "console=ttyLP1,115200 earlycon=lpuart32,0x5a070000,115200";
-		stdout-path = &lpuart1;
-	};
-
 	regulators {
 		compatible = "simple-bus";
 		#address-cells = <1>;
@@ -49,58 +42,7 @@
 
 &iomuxc {
 	pinctrl-names = "default";
-	imx8qxp-mek {
-
-		pinctrl_lpuart0: lpuart0grp {
-			fsl,pins = <
-				SC_P_UART0_RX_ADMA_UART0_RX	0x06000020
-				SC_P_UART0_TX_ADMA_UART0_TX	0x06000020
-			>;
-		};
-
-		pinctrl_lpuart1: lpuart1grp {
-			fsl,pins = <
-				SC_P_UART1_RX_ADMA_UART1_RX	0x06000020
-				SC_P_UART1_TX_ADMA_UART1_TX	0x06000020
-			>;
-		};
-
-		pinctrl_fec1: fec1grp {
-			fsl,pins = <
-				SC_P_ENET0_MDC_CONN_ENET0_MDC			0x06000048
-				SC_P_ENET0_MDIO_CONN_ENET0_MDIO			0x06000048
-				SC_P_ENET0_RGMII_TX_CTL_CONN_ENET0_RGMII_TX_CTL	0x06000048
-				SC_P_ENET0_RGMII_TXC_CONN_ENET0_RGMII_TXC	0x06000048
-				SC_P_ENET0_RGMII_TXD0_CONN_ENET0_RGMII_TXD0	0x06000048
-				SC_P_ENET0_RGMII_TXD1_CONN_ENET0_RGMII_TXD1	0x06000048
-				SC_P_ENET0_RGMII_TXD2_CONN_ENET0_RGMII_TXD2	0x06000048
-				SC_P_ENET0_RGMII_TXD3_CONN_ENET0_RGMII_TXD3	0x06000048
-				SC_P_ENET0_RGMII_RXC_CONN_ENET0_RGMII_RXC	0x06000048
-				SC_P_ENET0_RGMII_RX_CTL_CONN_ENET0_RGMII_RX_CTL	0x06000048
-				SC_P_ENET0_RGMII_RXD0_CONN_ENET0_RGMII_RXD0	0x06000048
-				SC_P_ENET0_RGMII_RXD1_CONN_ENET0_RGMII_RXD1	0x06000048
-				SC_P_ENET0_RGMII_RXD2_CONN_ENET0_RGMII_RXD2	0x06000048
-				SC_P_ENET0_RGMII_RXD3_CONN_ENET0_RGMII_RXD3	0x06000048
-			>;
-		};
-
-		pinctrl_fec2: fec2grp {
-			fsl,pins = <
-				SC_P_ESAI0_SCKR_CONN_ENET1_RGMII_TX_CTL		0x06000048
-				SC_P_ESAI0_FSR_CONN_ENET1_RGMII_TXC		0x06000048
-				SC_P_ESAI0_TX4_RX1_CONN_ENET1_RGMII_TXD0	0x06000048
-				SC_P_ESAI0_TX5_RX0_CONN_ENET1_RGMII_TXD1	0x06000048
-				SC_P_ESAI0_FST_CONN_ENET1_RGMII_TXD2		0x06000048
-				SC_P_ESAI0_SCKT_CONN_ENET1_RGMII_TXD3		0x06000048
-				SC_P_ESAI0_TX0_CONN_ENET1_RGMII_RXC		0x06000048
-				SC_P_SPDIF0_TX_CONN_ENET1_RGMII_RX_CTL		0x06000048
-				SC_P_SPDIF0_RX_CONN_ENET1_RGMII_RXD0		0x06000048
-				SC_P_ESAI0_TX3_RX2_CONN_ENET1_RGMII_RXD1	0x06000048
-				SC_P_ESAI0_TX2_RX3_CONN_ENET1_RGMII_RXD2	0x06000048
-				SC_P_ESAI0_TX1_CONN_ENET1_RGMII_RXD3		0x06000048
-			>;
-		};
-
+	imx8qxp-tqma8qx {
 		pinctrl_lpi2c1: lpi1cgrp {
 			fsl,pins = <
 				SC_P_USB_SS3_TC1_ADMA_I2C1_SCL	0x06000020
@@ -156,50 +98,6 @@
 			>;
 		};
 
-		pinctrl_usdhc2_gpio: usdhc2gpiogrp {
-			fsl,pins = <
-				SC_P_USDHC1_RESET_B_LSIO_GPIO4_IO19	0x06000021
-				SC_P_USDHC1_WP_LSIO_GPIO4_IO21		0x06000021
-				SC_P_USDHC1_CD_B_LSIO_GPIO4_IO22	0x06000021
-			>;
-		};
-
-		pinctrl_usdhc2: usdhc2grp {
-			fsl,pins = <
-				SC_P_USDHC1_CLK_CONN_USDHC1_CLK		0x06000041
-				SC_P_USDHC1_CMD_CONN_USDHC1_CMD		0x06000021
-				SC_P_USDHC1_DATA0_CONN_USDHC1_DATA0	0x06000021
-				SC_P_USDHC1_DATA1_CONN_USDHC1_DATA1	0x06000021
-				SC_P_USDHC1_DATA2_CONN_USDHC1_DATA2	0x06000021
-				SC_P_USDHC1_DATA3_CONN_USDHC1_DATA3	0x06000021
-				SC_P_USDHC1_VSELECT_CONN_USDHC1_VSELECT	0x06000021
-			>;
-		};
-
-		pinctrl_usdhc2_100mhz: usdhc2grp100mhz {
-			fsl,pins = <
-				SC_P_USDHC1_CLK_CONN_USDHC1_CLK		0x06000040
-				SC_P_USDHC1_CMD_CONN_USDHC1_CMD		0x06000020
-				SC_P_USDHC1_DATA0_CONN_USDHC1_DATA0	0x06000020
-				SC_P_USDHC1_DATA1_CONN_USDHC1_DATA1	0x06000020
-				SC_P_USDHC1_DATA2_CONN_USDHC1_DATA2	0x06000020
-				SC_P_USDHC1_DATA3_CONN_USDHC1_DATA3	0x06000020
-				SC_P_USDHC1_VSELECT_CONN_USDHC1_VSELECT	0x06000020
-			>;
-		};
-
-		pinctrl_usdhc2_200mhz: usdhc2grp200mhz {
-			fsl,pins = <
-				SC_P_USDHC1_CLK_CONN_USDHC1_CLK		0x06000040
-				SC_P_USDHC1_CMD_CONN_USDHC1_CMD		0x06000020
-				SC_P_USDHC1_DATA0_CONN_USDHC1_DATA0	0x06000020
-				SC_P_USDHC1_DATA1_CONN_USDHC1_DATA1	0x06000020
-				SC_P_USDHC1_DATA2_CONN_USDHC1_DATA2	0x06000020
-				SC_P_USDHC1_DATA3_CONN_USDHC1_DATA3	0x06000020
-				SC_P_USDHC1_VSELECT_CONN_USDHC1_VSELECT	0x06000020
-			>;
-		};
-
 		pinctrl_flexspi0: flexspi0grp {
 			fsl,pins = <
 				SC_P_QSPI0A_DATA0_LSIO_QSPI0A_DATA0	0x0600004c
@@ -223,18 +121,6 @@
 	};
 };
 
-&lpuart0 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_lpuart0>;
-	status = "disabled";
-};
-
-&lpuart1 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_lpuart1>;
-	status = "okay";
-};
-
 &gpio0 {
 	status = "okay";
 };
@@ -251,41 +137,6 @@
 	status = "okay";
 };
 
-&fec1 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_fec1>;
-	phy-mode = "rgmii";
-	phy-handle = <&ethphy0>;
-	fsl,ar8031-phy-fixup;
-	fsl,magic-packet;
-	status = "disabled";
-
-	mdio {
-		#address-cells = <1>;
-		#size-cells = <0>;
-
-		ethphy0: ethernet-phy@0 {
-			compatible = "ethernet-phy-ieee802.3-c22";
-			reg = <0>;
-		};
-
-		ethphy1: ethernet-phy@1 {
-			compatible = "ethernet-phy-ieee802.3-c22";
-			reg = <1>;
-		};
-	};
-};
-
-&fec2 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_fec2>;
-	phy-mode = "rgmii";
-	phy-handle = <&ethphy1>;
-	fsl,ar8031-phy-fixup;
-	fsl,magic-packet;
-	status = "okay";
-};
-
 &flexspi0 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_flexspi0>;
@@ -319,25 +170,3 @@
 	non-removable;
 	status = "okay";
 };
-
-&usdhc2 {
-	pinctrl-names = "default", "state_100mhz", "state_200mhz";
-	pinctrl-0 = <&pinctrl_usdhc2>, <&pinctrl_usdhc2_gpio>;
-	pinctrl-1 = <&pinctrl_usdhc2_100mhz>, <&pinctrl_usdhc2_gpio>;
-	pinctrl-2 = <&pinctrl_usdhc2_200mhz>, <&pinctrl_usdhc2_gpio>;
-	bus-width = <4>;
-	cd-gpios = <&gpio4 22 GPIO_ACTIVE_LOW>;
-	wp-gpios = <&gpio4 21 GPIO_ACTIVE_HIGH>;
-	vmmc-supply = <&reg_usdhc2_vmmc>;
-	status = "okay";
-};
-
-&usbotg1 {
-	srp-disable;
-	hnp-disable;
-	adp-disable;
-	power-polarity-active-high;
-	disable-over-current;
-	status = "disabled";
-};
-
-- 
2.17.1

